service/src/main/java/uk/gov/hmcts/reform/fpl/service/DraftCMOService.java

```diff
@@ -42,18 +48,93 @@ public DraftCMOService(DateFormatterService dateFormatterService, ObjectMapper m
         this.directionHelperService = directionHelperService;
     }
 
-    public DynamicList getHearingDateDynamicList(CaseDetails caseDetails) {
-        Map<String, Object> data = caseDetails.getData();
-        CaseData caseData = mapper.convertValue(data, CaseData.class);
+    public Map<String, Object> extractIndividualCaseManagementOrderObjects(
+        CaseManagementOrder caseManagementOrder,
+        List<Element<HearingBooking>> hearingDetails) {
 
-        List<Element<HearingBooking>> hearingDetails = caseData.getHearingDetails();
+        if (isNull(caseManagementOrder)) {
+            caseManagementOrder = CaseManagementOrder.builder().build();
+        }
+
+        HashMap<String, Object> data = new HashMap<>();
+        HashMap<String, Object> reviewCaseManagementOrder = new HashMap<>();
```

Comment:
Please do not operate on maps. These are harder to refactor, more difficult to read and lead to bad code.